src/bitcoin-cli.cpp

```diff
@@ -235,7 +235,21 @@ UniValue CallRPC(const std::string& strMethod, const UniValue& params)
     assert(output_buffer);
     evbuffer_add(output_buffer, strRequest.data(), strRequest.size());
 
-    int r = evhttp_make_request(evcon.get(), req.get(), EVHTTP_REQ_POST, "/");
+    // check if we should use a special wallet endpoint
+    std::string endpoint = "/";
+    std::string walletName = GetArg("-wallet", "");
```

Comment:
This is still not addressed. If you're going to add extra options to bitcoin-cli, please also update the help text.

Strongly recommend this change goes in its own commit.